cmd/kubelet/app/options/options.go

```diff
@@ -519,6 +514,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.StringSliceVar(&c.AllowedUnsafeSysctls, "allowed-unsafe-sysctls", c.AllowedUnsafeSysctls, "Comma-separated whitelist of unsafe sysctls or unsafe sysctl patterns (ending in *). Use these at your own risk.")
 
 	fs.Int32Var(&c.NodeStatusMaxImages, "node-status-max-images", c.NodeStatusMaxImages, "The maximum number of images to report in Node.Status.Images. If -1 is specified, no cap will be applied.")
+	fs.BoolVar(&c.KernelMemcgNotification, "kernel-memcg-notification", c.KernelMemcgNotification, "If enabled, the kubelet will integrate with the kernel memcg notification to determine if memory eviction thresholds are crossed rather than polling.")
```

Comment:
We should keep the flag name the same for backwards compatibility during the deprecation period.
One technique to manage this is to add a new flag called "kernel-memcg-notification", but still keep the old one and just mark the old one deprecated. BootstrapKubeconfig is one example of that (in this case both can point to the same field in the config, instead of flags struct, but the concept is the same).

Reply:
I had to bring this flag to `AddKubeletConfigFlags` and reuse the KubeletConfig from there,  by default it adds the deprecation notes, LMK if it works?

```
--experimental-kernel-memcg-notification                   Use kernelMemcgNotification configuration. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
--kernel-memcg-notification                                If enabled, the kubelet will integrate with the kernel memcg notification to determine if memory eviction thresholds are crossed rather than polling. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
```

Reply:
curious, why the name change (why remove the "experimental" part)?

Reply:
technically it "graduated", and is not experimental anymore, we are keeping the --experimental prefixed one for backward compatibility, it must be removed in a few releases.